### ISSUE TITLE
[Guided Setup Stage 2] Update HandleSpecialChatReportsOnMergeAccount to handle system DM

### ISSUE THREAD

**Issue by francoisl** [Expensify/App#38770]
## Part of the "wave-collect – Build Stage 2 of Guided Setup" project
Main issue: https://github.com/Expensify/Expensify/issues/356685
Doc section: https://docs.google.com/document/d/10bhCv6XtzzqEZQ9tzxFwAz4lowgjOhsFqz13BMq6Iu4/edit#heading=h.dmxmb440p80k (and section below)
Project: https://github.com/orgs/Expensify/projects/129

### Feature Description
- Create the concept of read-only messages by updating the permissions mechanism
- Create a read-only system DM using the new Expensify persona for existing and new users

### Remaining tasks 

- [x] ~Make `Account writing to chat/task report without permission` throw~
- [ ] Handle MergeAccount
    - [ ] Update HandleSpecialChatReportsOnMergeAccount to handle system DM
    - [x] Remove HandleSelfDMOnMergeAccount from AuthCommandMap

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010dca0075d6efe79d</li>
        <li>Upwork Job ID: 1770904803490025472</li>
        <li>Last Price Increase: 2024-03-21</li>
    </ul>
</details>

**Comment by francoisl** [comment]
This is all backend work and will be taken care of internally, no need for a C+ here.

**Comment by trjExpensify** [comment]
Still need a volunteer here, Melv.

**Comment by flodnv** [comment]
Taking this! @francoisl I'm assuming I need to work on
2. Creating the system DM (phase 2)
3. Enforcing read-only permissions for the system DM (phase 2)

?

**Comment by flodnv** [comment]
- Added a few doc comments on these 2 sections
- Pushed WIP PR
- Paused at `Finally, we'll need to make a few general changes to accommodate system DMs:`, will resume once we figure out answers to my questions

**Comment by flodnv** [comment]
>Account writing to chat/task report without permission.

is logged quite a bit... https://www.expensify.com/_devportal/tools/logSearch/#sort=asc&size=10000&query=blob%3A%22Account%20writing%20to%20chat%2Ftask%20report%20without%20permission.%22%20AND%20timestamp%3A%5B2024-04-23T00%3A00%20TO%202024-04-24T23%3A59%5D



---

Example 1


>Trying to create action (reportID 1500222615990510, requestingAccountID: 15751631, actionAccountID: 15751631, action: ACTIONABLEJOINREQUEST, reportActionID: 0)

https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%22878be8ca28e85a6d-MEL%22)+AND+timestamp:[2024-04-23T05:48:41.502Z+TO+2024-04-23T07:48:41.502Z]&index=logs_expensify-027972

#admins room 

Wild guess: join requests are made by non admins, and for some reason we're using that non-admin accountID to create the action in the #admins room 

--- 

Example 2 

>Trying to create action (reportID 7411466038216825, requestingAccountID: 8392101, actionAccountID: 8392101, action: ACTIONABLETRACKEXPENSEWHISPER, reportActionID: 0)

https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%2287900eff7a727759-LHR%22)+AND+timestamp:[2024-04-23T17:53:49.505Z+TO+2024-04-23T19:53:49.505Z]&index=logs_expensify-027982

self DM

Wild guess: for some reason this is failing to see that the owner has the right to write to the report 😕 

---

I'll leave this alone for now, implement what's in the doc, and then get back to this

**Comment by flodnv** [comment]
Most (if not everything) of what's under `Finally, we'll need to make a few general changes to accommodate system DMs:` in the doc is on HOLD for code written in @NikkiWines's PR https://github.com/Expensify/Auth/pull/10585

**Comment by flodnv** [comment]
@francoisl just confirming: once @NikkiWines's PR is merged, I can focus on finishing the whole `Enforcing read-only permissions for the system DM (phase 2)` section, and I do NOT do `Creating an API command to set up the initial tasks  (phase 1)`, correct?

**Comment by francoisl** [comment]
Yes that sounds good to me Flo! Thanks for the help.

**Comment by francoisl** [comment]
> Example 1
>
> #admins room
>
> Wild guess: join requests are made by non admins, and for some reason we're using that non-admin accountID to create the action in the #admins room

Yes that looks right according to the code in the command `RequestPolicyAccess`. Going to look into the rest and review your PR now.

**Comment by flodnv** [comment]
Example 3: https://expensify.slack.com/archives/C03U7DCU4/p1713968604674249

**Comment by flodnv** [comment]
Example 4 repro steps:
1. Create a report with some expenses
2. Submit report to someone on the policy who has role=user
3. As that person, reject a transaction on the processing report 💥

**Comment by flodnv** [comment]
PR for example 4 ^

**Comment by flodnv** [comment]
- The log `Account writing to chat/task report without permission` has quiet down. Let's wait 1 week to see if we can throw and undo https://github.com/Expensify/Auth/pull/10687/files#diff-7d3f89cee9578c11d06f3f40aae607662648d2aab066d79e6b14d73bb0c7ff9aR110
- I still have a few items to complete from the design doc, shouldn't take too long.

**Comment by flodnv** [comment]
I looked at the remaining 3 tasks today and left some questions on each one of them for @francoisl

**Comment by francoisl** [comment]
Thanks! Left comments in the doc.

**Comment by flodnv** [comment]
Spent all my day on a fire

**Comment by flodnv** [comment]
`Account writing to chat/task report without permission` shows up in 2 more flows:
1. EditTask
2. Policy_Delete -> archive reports?

**Comment by flodnv** [comment]
Sent 2 PRs to rename command to `HandleSpecialChatReportsOnMergeAccount` ^

I updated the OP with the remaining tasks as I was getting a bit lost in the weeds

**Comment by flodnv** [comment]
From the OP:

> Update code so that system DM header shows as Expensify

@trjExpensify just brought to my attention that you're already working on that here https://github.com/Expensify/App/pull/41290 ?

Can you please confirm that the 2 other remaining items in the OP are not being worked on by anyone else, and they're mine to complete? cc @NikkiWines

**Comment by NikkiWines** [comment]
The other two items don't seem to overlap, so you're all set 👍 

As for ` Update code so that system DM header shows as Expensify`, can't we do that by manually setting the personal details for `notifications@`? Though maybe we programmatically set the concierge details 🤔

**Comment by flodnv** [comment]
No, I tested and that does not work, we need @francoisl's change here: https://github.com/Expensify/App/pull/41290/files#diff-577fcc5f3a5e4930916ff310e61c38edfa093792fe3ab55b9a30d2aecd8811dbR3163

**Comment by flodnv** [comment]
No update, I was focused on the fire.

**Comment by flodnv** [comment]
> Make Account writing to chat/task report without permission throw

- User_DowngradeToFreeAccount : https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A(%22883ab0d4adbd6789-ATL%22)%20AND%20timestamp%3A%5B2024-05-14T10%3A54%3A14.580Z%20TO%202024-05-14T12%3A54%3A14.580Z%5D&index=_all
- ScrapeCard: https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A(%22883bde2d9aa17104-YVR%22)%20AND%20timestamp%3A%5B2024-05-14T14%3A19%3A43.105Z%20TO%202024-05-14T16%3A19%3A43.105Z%5D&index=_all
- RejectTransaction: https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A(%22883beceb4dfc074b-MAN%22)%20AND%20timestamp%3A%5B2024-05-14T14%3A29%3A44.472Z%20TO%202024-05-14T16%3A29%3A44.472Z%5D&index=_all

**Comment by flodnv** [comment]
I've been focusing on the fire. Feel free to take this from me @francoisl or @NikkiWines 😅

**Comment by flodnv** [comment]
No update, focused on fire and comp

**Comment by flodnv** [comment]
I'm switching my top focus to #fast-apis and will be ooo next week so I'm unassigning myself from this.

**Comment by trjExpensify** [comment]
@francoisl @NikkiWines confirming this hasn't been taken care of elsewhere?

>Update HandleSpecialChatReportsOnMergeAccount to handle system DM

If not, can we make sure someone takes it to complete that to wrap-up this project.

**Comment by NikkiWines** [comment]
I don't believe this has been taken care of yet, no

**Comment by trjExpensify** [comment]
Okay got it, who fancies nabbing it? Don't all jump at once.. 😂

**Comment by trjExpensify** [comment]
Going back to `weekly`, hoping @NikkiWines might be able to get to it this week.

**Comment by NikkiWines** [comment]
Discussing with @francoisl today about whether or not we want to update this command to 

A. Take in the type of special chat report we'd want to merge (i.e. system, selfDM, etc.) 

or 

B. Merge all the special type DMs in one go instead of passing a specific type. 

Not sure if we have a scenario where we'd want to merge the selfDM but not system DM, since this is called when the user merges their account. So I'm leaning more towards B at the moment

**Comment by francoisl** [comment]
I'd say option `B`, otherwise we'd need that command twice (once for self DMs, once for system DMs).

**Comment by NikkiWines** [comment]
PR is in the works! Draft is up [here](https://github.com/Expensify/Auth/pull/12482)

**Comment by NikkiWines** [comment]
Logic is working as expected, finishing up tests tomorrow morning

**Comment by NikkiWines** [comment]
PR is in review!

**Comment by NikkiWines** [comment]
PR has been merged!

**Comment by trjExpensify** [comment]
Dope, deployed. Good to close?

**Comment by NikkiWines** [comment]
Yes should be all set here!

**Comment by trjExpensify** [comment]
Wahoo!
